nat-lab/tests/utils/testing.py

```diff
@@ -10,8 +12,102 @@ def unpack_optional(opt: Optional[T]) -> T:
     return opt
 
 
-def test_name_safe_for_file_name():
+def get_current_test_file_name() -> str | None:
+    """
+    Returns a string representing the filename of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+
+    For example:
+    "tests/test_mesh_plus_vpn.py" -> "test_mesh_plus_vpn"
+    """
+    test_name = os.environ.get("PYTEST_CURRENT_TEST")
+    if test_name is not None:
+        try:
+            # take the file path part of the pytest string, remove extension and parent directory
+            file_part = test_name.split("::")[0].split(".")[0].split("/")[-1]
+            return format_path_string(file_part)
+        except IndexError:
+            print(datetime.now(), "get_current_test_file_name() IndexError", test_name)
+            return test_name
+    print(datetime.now(), "PYTEST_CURRENT_TEST is None")
+    return None
+
+
+def get_current_test_case_name() -> str | None:
+    """
+    Returns a string representing the case name of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+
+    For example:
+    "tests/test_mesh_plus_vpn.py::test_vpn_plus_mesh_over_direct" -> "test_vpn_plus_mesh_over_direct"
+    """
     test_name = os.environ.get("PYTEST_CURRENT_TEST")
     if test_name is not None:
-        return "".join([x if x.isalnum() else "_" for x in test_name.split(" ")[0]])
-    return test_name
+        try:
+            test_part = test_name.split("::", 1)[-1].split(" ")[0].split("[", 1)[0]
+            return format_path_string(test_part)
+        except IndexError:
```

Comment:
Personally I would not try to catch it and I would let it fail so we know we have some edge cases to cover early

Reply:
But do we want to fail the whole pipeline in case the logs are not formatted correctly?

Reply:
Why not. It will literally never happen. Unless in 5 years someone in pytest goes crazy and changes this format. Then we want to catch it and fix this place :)

Reply:
Good point!